code/__HELPERS/_lists.dm

```diff
@@ -177,7 +177,11 @@
 		result = first ^ second
 	return result
 
-//Pretends to pick an element based on its weight but really just seems to pick a random element.
+//Picks a random element from a list based on a weighting system:
+//1. Adds up the total of weights for each element
+//2. Gets a number between 1 and that total
+//3. For each element in the list, subtracts its weighting from that number
+//4. If that makes the number 0 or less, return that element.

```

Comment:
it's not a useful comment if you can easily see what the code does by looking at it.